current-time/current-time.test.ts

```diff
@@ -0,0 +1,169 @@
+// @ts-nocheck
+import { createEvent } from 'effector';
+import { toBeCloseWithThreshold, wait } from '../test-library';
+import { currentTime } from './index';
+
+const TIMER_THRESHOLD = 70;
+
+expect.extend({ toBeCloseWithThreshold });
+
+describe('start/stop', () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  test('should not start automatically', () => {
+    const start = createEvent();
+    const stop = createEvent();
+    const $now = currentTime({ start, stop });
+
+    const watcher = jest.fn();
+    $now.updates.watch(watcher);
+
+    jest.runAllTimers();
+
+    expect(watcher).not.toBeCalled();
+  });
+
+  test('should start after start event called', () => {
+    const start = createEvent();
+    const stop = createEvent();
+    const $now = currentTime({ start, stop });
+
+    const watcher = jest.fn();
+    $now.updates.watch(watcher);
+
+    start();
+    jest.advanceTimersByTime(1000);
+    stop();
+
+    expect(watcher).toBeCalled();
+  });
+
+  test('should stop after stop event called', () => {
+    const start = createEvent();
+    const stop = createEvent();
+    const $now = currentTime({ start, stop });
+
+    const watcher1 = jest.fn();
+    $now.updates.watch(watcher1);
+
+    start();
+    jest.advanceTimersByTime(1000);
+    stop();
+
+    const watcher2 = jest.fn();
+    $now.updates.watch(watcher2);
+
+    expect(watcher1).toBeCalled();
+    expect(watcher2).not.toBeCalled();
+  });
+
+  test('should restart after start-stop-start event chain', () => {
+    const start = createEvent();
+    const stop = createEvent();
+    const $now = currentTime({ start, stop });
+
+    start();
+    jest.advanceTimersByTime(1000);
+    stop();
+
+    const watcher = jest.fn();
+    $now.updates.watch(watcher);
+
+    start();
+    jest.advanceTimersByTime(1000);
+    stop();
+
+    expect(watcher).toBeCalled();
+  });
+});
+
+describe('interval', () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  test('should use default interval for undefined interval', () => {
+    const start = createEvent();
+    const stop = createEvent();
+    const $now = currentTime({ start, stop });
+
+    const watcher = jest.fn();
+    $now.updates.watch(watcher);
+
+    start();
+
+    // default interval is 100ms
+    // run 10 updates
+    jest.advanceTimersByTime(1000);
+
+    stop();
+
+    // 11 = 10 updates + 1 update on start
+    expect(watcher).toBeCalledTimes(11);
+  });
+
+  test('should use custom interval', () => {
+    const testIntrevals = [20, 200, 2000];
+
+    for (const interval of testIntrevals) {
```

Comment:
You can use [`test.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) to simplify test:

```suggestion
  test.each([20, 200, 2000])('should use %i interval', (interval) => {
```